sdk/lib/opentelemetry/sdk/trace/tracer_provider.rb

```diff
@@ -138,7 +138,7 @@ def internal_start_span(name, kind, attributes, links, start_timestamp, parent_c
           trace_id ||= @id_generator.generate_trace_id
           result = @sampler.should_sample?(trace_id: trace_id, parent_context: parent_context, links: links, name: name, kind: kind, attributes: attributes)
           span_id = @id_generator.generate_span_id
-          if result.recording? && !@stopped
+          if !OpenTelemetry::Common::Utilities.untraced?(parent_context) && result.recording? && !@stopped
```

Comment:
IMO we should return a non-recording span without calling `@sampler.should_sample` if `untraced?(parent_context)`.

Rationale:
1. We should not create a non-recording span using the tracestate from a sampler whose result we're overriding.
2. We should re-use the span ID from the parent span (if there is one) to avoid breaking the trace (in case we propagate the trace context with an outbound request and the server makes it own sampling decision).

Reply:
like this? 5f9b4a282039442088e5f6150a3273a6ee4ec0da